content/en/docs/concepts/storage/persistent-volumes.md

```diff
@@ -227,8 +227,17 @@ kubectl describe pvc <pvc_name>
 
 If the `PersistentVolumeClaim` has the status `FileSystemResizePending`, it is safe to recreate the pod using the PersistentVolumeClaim.
 
+FlexVolumes allow resize if the driver is set with the `RequiresFSResize` capability to true. 
+The FlexVolume can be resized on pod restart as 'ExpandPersistentVolume' feature gate is enabled by default. 
+
 #### Resizing an in-use PersistentVolumeClaim
 
+Expanding in-use PVCs for FlexVolumes is added in release 1.13. To enable this feature use  `ExpandInUsePersistentVolumes` and `ExpandPersistentVolumes` feature gates. The `ExpandPersistentVolumes` feature gate is already enabled by default. If the `ExpandInUsePersistentVolumes` is set, FlexVolume can be resized online without pod restart. 
+ 
+The FlexVolume plugin uses RequiresFSResize to identify and resize the Persistent Volume. The underlying driver must support resize, if not an error will be returned and Persistent Volume object will not change and the expand request will fail. FlexVolume driver must implement the method `ExpandFS` defined in the interface `FSResizableVolumePlugin` along with `ExpandableVolumePlugin` interface and its method `ExpandVolumeDevice`.
+The Flex driver should implement ExpandableVolumePlugin interface only if it is to be installed on the controller/master.
+The Flex driver should implement any volume expansion methods along with FS resize in ExpandFS method if the Flex driver is installed on the worker nodes but not on the controller or master node.
+
```

Comment:
IMO - we should delete this entire section. There isn't anything special about flex that needs documenting here. Flex specific implementation details should go in document I linked below.

Reply:
I think it's probably important to keep the sentence "The underlying driver must support resize" since that there's a chance administrators might be confused if this isn't working for the drivers they have deployed.

Reply:
partly agree but at very least it should go below previous text. This section is about expanding in-use persistent volumes and not just flex volume, so there is no need to document this first thing.

Reply:
I added a not to satisfy this, please review.

Reply:
Thanks. But what I meant was - `Resizing an in-use PersistentVolumeClaim` section introduces expansion of in-use volumes regardless of plugin type. This section is not just about flex. But currently you have put flex item as first thing under `in-use` section. So if you read rendered markdown, it reads weird. 

```
Resizing an in-use PersistentVolumeClaim

Expanding in-use PVCs for FlexVolumes is added in release 1.13. To enable this feature use ExpandInUsePersistentVolumes and ExpandPersistentVolumes feature gates. The ExpandPersistentVolumes feature gate is already enabled by default. If the ExpandInUsePersistentVolumes is set, FlexVolume can be resized online without pod restart.

{{< note >}} Note: FlexVolume resize is possible only when the underlying driver supports resize. {{< /note >}}

{{< feature-state for_k8s_version="v1.11" state="alpha" >}}

Expanding in-use PVCs is an alpha feature. To use it, enable the ExpandInUsePersistentVolumes feature gate. In this case, you don't need to delete and recreate a Pod or deployment that is using an existing PVC. Any in-use PVC automatically becomes available to its Pod as soon as its file system has been expanded. This feature has no effect on PVCs that are not in use by a Pod or deployment. You must create a Pod which uses the PVC before the expansion can complete.

{{< note >}} Note: Expanding EBS volumes is a time consuming operation. Also, there is a per-volume quota of one modification every 6 hours. {{< /note >}}
```

This is how the whole text reads as. Whereas what I was suggesting is - this diff should not alter first paragraph of the section. The blurb about flexvolume can be added as second paragraph rather than sitting on top of the section.  Hope that clarifies it.

Reply:
This is how I propose the section should read:

Expanding in-use PVCs is an alpha feature. To use it, enable the ExpandInUsePersistentVolumes feature gate. In this case, you don't need to delete and recreate a Pod or deployment that is using an existing PVC. Any in-use PVC automatically becomes available to its Pod as soon as its file system has been expanded. This feature has no effect on PVCs that are not in use by a Pod or deployment. You must create a Pod which uses the PVC before the expansion can complete.

Expanding in-use PVCs for FlexVolumes is added in release 1.13. To enable this feature use ExpandInUsePersistentVolumes and ExpandPersistentVolumes feature gates. The ExpandPersistentVolumes feature gate is already enabled by default. If the ExpandInUsePersistentVolumes is set, FlexVolume can be resized online without pod restart.
{{< note >}} Note: FlexVolume resize is possible only when the underlying driver supports resize. {{< /note >}}

{{< feature-state for_k8s_version="v1.11" state="alpha" >}}

{{< note >}} Note: Expanding EBS volumes is a time consuming operation. Also, there is a per-volume quota of one modification every 6 hours. {{< /note >}}

Reply:
got it, thanks for the review.